src/openLibrary/producer/parser.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2018  Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import {entityTypes, isNotDefined, sortName} from '../../helpers/utils';
+import {identifiers, mapLanguage} from '../../helpers/mapping';
+import _ from 'lodash';
+import franc from 'franc-min';
+
+
+const WORK = 'work';
+const EDITION = 'edition';
+const AUTHOR = 'author';
+
+
+function detectLanguage(name) {
+	let lang = franc(name);
+	lang = lang !== 'und' ? lang : 'eng';
+
+	return mapLanguage(lang);
+}
+
+function processWork(json) {
+	if (isNotDefined(json)) {
+		return null;
+	}
+
+	const work = {};
+	work.alias = [];
+	work.metadata = {};
+	work.metadata.relationships = [];
+	work.metadata.links = [];
+	work.identifiers = [];
+
+	work.entityType = entityTypes.WORK;
+	work.source = 'OPENLIBRARY';
+
+	// Set up aliases
+	// The first alias (in the logical flow) is made to be default alias
+	let defaultAlias = true;
+	if (!isNotDefined(json.title)) {
+		const name = json.title;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			primary: true,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+	if (!isNotDefined(json.subtitle)) {
+		const name = json.subtitle;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+
+
+	// Last Edited at source
+	if (!isNotDefined(_.get(json, 'last_modified.value'))) {
+		work.lastEdited = json.last_modified.value;
+	}
+
+	if (!isNotDefined(json.key)) {
+		const openLibraryWorkId = json.key.split('/')[2];
+		work.identifiers.push({
+			typeId: identifiers.openLibraryWorkId,
+			value: openLibraryWorkId
+		});
+		work.originId = openLibraryWorkId;
+	}
+
+	if (!isNotDefined(json.authors) && (json.authors instanceof Array)) {
+		json.authors.forEach((authorObj) => {
+			if (!isNotDefined(authorObj && _.get(authorObj, 'author.key'))) {
+				work.metadata.relationships.push({
+					type: 'authoredBy',
+					value: authorObj.author.key.split('/')[2]
+				});
+			}
+		});
+	}
+
+	if (!isNotDefined(json.description)) {
```

Comment:
The thinking behind doing the above was - right now there's no way to display the annotation - so when user clicks on edit and approve s/he can't see the annotation. On the other hand, imported disambiguation shows up and they can edit it if required. 
Right now the present model works with saving the imported disambiguation when an editor approves. In case we decide to have the user input the approve time disambiguation, then we will have to add a separate page with a form for that.